File: api/src/test/java/net/blueshell/api/factory/dto/user/SimpleUserDTOFactory.java
package net.blueshell.api.factory.dto.user;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.user.SimpleUserDTO;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for SimpleUserDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class SimpleUserDTOFactory extends BaseDtoFactory<SimpleUserDTO> {

    @Override
    public Class<SimpleUserDTO> targetType() {
        return SimpleUserDTO.class;
    }

    @Override
    public SimpleUserDTO createBasic() {
        SimpleUserDTO dto = new SimpleUserDTO();
        dto.setInitials("TU");
        dto.setFirstName("Test");
        dto.setLastName("User");
        dto.setPrefix(null);
        dto.setUsername(unique("user"));
        dto.setNewsletter(false);
        dto.setDiscord("testuser");
        dto.setEmail(email("user"));
        dto.setPhoneNumber("+31612345678");
        dto.setPassword("Password123!");
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/user/SimpleUserDTOFactory.kt
package net.blueshell.api.factory.dto.user

import net.blueshell.api.user.dto.SimpleUserDTO
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for SimpleUserDTO test instances.
 */
@Component
class SimpleUserDTOFactory : BaseDtoFactory<SimpleUserDTO>() {

    override fun targetType(): Class<SimpleUserDTO> = SimpleUserDTO::class.java

    override fun createBasic(): SimpleUserDTO {
        val dto = SimpleUserDTO()
        dto.initials = "TU"
        dto.firstName = "Test"
        dto.lastName = "User"
        dto.prefix = null
        dto.username = unique("user")
        dto.newsletter = false
        dto.discord = "testuser"
        dto.email = email("user")
        dto.phoneNumber = "+31612345678"
        dto.password = "Password123!"
        return dto
    }
}
